prog/vnet/nic_nexus.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::NicNexus < Prog::Base
+  semaphore :destroy, :refresh_mesh, :detach_vm
+
+  def self.assemble(project_id, private_subnet_id, name: nil, ipv6_addr: nil, ipv4_addr: nil)
+    project = Project[project_id]
+    unless project || Config.development?
+      fail "No existing project"
+    end
+
+    # maybe wait vnet
+    subnet = PrivateSubnet[private_subnet_id]
+    unless subnet
+      fail "Given subnet doesn't exist with the id #{private_subnet_id}"
+    end
+
+    ubid = UBID.generate(UBID::TYPE_NIC)
+    tap_name = Nic.ubid_to_tap_name(ubid)
+
+    Validation.validate_name(tap_name)
```

Comment:
Not sure if we need this, since this is not a value entered by user (unlike the `name` in `Nexus.assemble` which could be user entered)

Did you mean to validate the "name" parameter instead?

Also what happens if `name` is `nil`? Do we keep is as `nil` or `||` with the auto-generated `tap_name` similar to `Nexus.assemble`?